qa/full-cluster-restart/src/test/java/org/elasticsearch/upgrades/FullClusterRestartIT.java

```diff
@@ -1386,67 +1390,97 @@ public void testResize() throws Exception {
         }
     }
 
-    public void testCreateSystemIndexInOldVersion() throws Exception {
-        assumeTrue("only run on old cluster", isRunningAgainstOldCluster());
-        // create index
-        Request createTestIndex = new Request("PUT", "/test_index_old");
-        createTestIndex.setJsonEntity("{\"settings\": {\"index.number_of_replicas\": 0}}");
-        client().performRequest(createTestIndex);
-
-        Request bulk = new Request("POST", "/_bulk");
-        bulk.addParameter("refresh", "true");
-        bulk.setJsonEntity("{\"index\": {\"_index\": \"test_index_old\"}}\n" +
-            "{\"f1\": \"v1\", \"f2\": \"v2\"}\n");
-        client().performRequest(bulk);
-
-        // start a async reindex job
-        Request reindex = new Request("POST", "/_reindex");
-        reindex.setJsonEntity(
-            "{\n" +
-                "  \"source\":{\n" +
-                "    \"index\":\"test_index_old\"\n" +
-                "  },\n" +
-                "  \"dest\":{\n" +
-                "    \"index\":\"test_index_reindex\"\n" +
-                "  }\n" +
-                "}");
-        reindex.addParameter("wait_for_completion", "false");
-        Map<String, Object> response = entityAsMap(client().performRequest(reindex));
-        String taskId = (String) response.get("task");
-
-        // wait for task
-        Request getTask = new Request("GET", "/_tasks/" + taskId);
-        getTask.addParameter("wait_for_completion", "true");
-        client().performRequest(getTask);
-
-        // make sure .tasks index exists
-        assertBusy(() -> {
+    @SuppressWarnings("unchecked")
+    public void testSystemIndexMetadataIsUpgraded() throws Exception {
+        if (isRunningAgainstOldCluster()) {
```

Comment:
Why have we combined two tests into one, where before they were using `assumeTrue` / `assumeFalse`?

Reply:
The code is logically the same, but combines the setup and check steps into one test method. 

Using `assumeTrue`/`assumeFalse` causes the assuming test method to effectively be skipped when the assumed condition is not met - and in this case, they weren't really separate test cases, but a setup step (on the old cluster) and a check step (on the new cluster). By combining them into one test method that branches based on which cluster it's running against, it makes it much easier to run the entire test, as you can now use `-Dtests.method=testSystemIndexMetadataIsUpgraded` instead of having to specify multiple methods.